controllers/storagecluster/cephblockpools.go

```diff
@@ -18,11 +18,22 @@ type ocsCephBlockPools struct{}
 // newCephBlockPoolInstances returns the cephBlockPool instances that should be created
 // on first run.
 func (r *StorageClusterReconciler) newCephBlockPoolInstances(initData *ocsv1.StorageCluster) ([]*cephv1.CephBlockPool, error) {
+	mirroringPeerSpec := &cephv1.MirroringPeerSpec{}
 	var mirroringSpec cephv1.MirroringSpec
+
 	if initData.Spec.Mirroring.Enabled {
 		mirroringSpec.Enabled = true
 		mirroringSpec.Mode = "image"
+
+		// ensures that peer cluster secret is exist and appending into cephBlockPool
+		secretName := initData.Name + "-" + "token"
+		_, err := r.retrieveSecret(secretName, initData)
+		if err == nil {
```

Comment:
if it fails we need to log the error. we need to know when we try to get a secret and it fails.

Reply:
done with log:
{"level":"error","ts":1629308702.1485183,"logger":"controllers.StorageCluster","msg":"Unable to append cluster peer token with CephBlockPool.","Request.Namespace":"openshift-storage","Request.Name":"ocs-storagecluster","CephBlockPool":"openshift-storage/ocs-storagecluster-cephblockpool","error":"Secret \"cluster-peer-token-ocs-storagecluster\" not found","stacktrace":"github.com/openshift/ocs-operator/controllers/storagecluster.(*ocsCephBlockPools).ensureCreated\n\t/home/gshanmug/Documents/new_123/src/github.com/openshift/upgrade/ocs-operator/controllers/storagecluster/cephblockpools.go:73\ngithub.com/openshift/ocs-operator/controllers/storagecluster.(*StorageClusterReconciler).reconcilePhases\n\t/home/gshanmug/Documents/new_123/src/github.com/openshift/upgrade/ocs-operator/controllers/storagecluster/reconcile.go:366\ngithub.com/openshift/ocs-operator/controllers/storagecluster.(*StorageClusterReconciler).Reconcile\n\t/home/gshanmug/Documents/new_123/src/github.com/openshift/upgrade/ocs-operator/controller...